src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java

```diff
@@ -120,8 +121,11 @@ public FormDataDefinition saveFormData(FormDataSchema formDataSchema) throws IOE
                 }
                 else
                 {
-                    Query query=new Query(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()));
+                    Query query=StringUtils.isNotBlank(filter)?new Query(new Criteria().andOperator(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()),
+                            getCriteria(filter))):new Query(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()));
                     Optional<FormDataDefinition> existingFormDataDefinition=Optional.ofNullable(mongoTemplate.findOne(query,FormDataDefinition.class,TP_RUNTIME_FORM_DATA + formId));
+                    Objects.requireNonNull(formDataDefinition).setCreatedById(existingFormDataDefinition.orElseThrow().getCreatedById());
```

Comment:
what happens if the form data definition is not found?

Reply:
checking for not null conditions and throwing 400 bad request for invalid input